src/Npgsql.NodaTime/TimestampTzHandler.cs

```diff
@@ -44,12 +45,21 @@ ZonedDateTime INpgsqlSimpleTypeHandler<ZonedDateTime>.Read(NpgsqlReadBuffer buf,
                     throw new NpgsqlSafeReadException(new NotSupportedException("Infinity values not supported for timestamp with time zone"));
                 return TimestampHandler.Decode(value).InZone(_dateTimeZoneProvider[buf.Connection.Timezone]);
             }
-            catch (TimeZoneNotFoundException) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
+            catch (Exception e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
             {
-                throw new NpgsqlSafeReadException(
-                    new TimeZoneNotFoundException(
-                        "The special PostgreSQL timezone 'localtime' is not supported when reading values of type 'timestamp with time zone'. " +
-                        "Please specify a real timezone in 'postgresql.conf' on the server, or set the 'PGTZ' environment variable on the client."));
+                switch (e)
+                {
+                case DateTimeZoneNotFoundException _:
+                case TimeZoneNotFoundException _:
```

Comment:
This can be simplified to:

```csharp
catch (Exception e) when (
    (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase)) &&
    (e is TimeZoneNotFoundException || e is DateTimeZoneNotFoundException))
{
    throw new NpgsqlSafeReadException(/* ... */);
}
```

Reply:
I personally like this one best.